tests/test_operator.py

```diff
@@ -203,6 +203,50 @@ def test_arguments(self):
         op4 = switchconfig(language='openmp')(Operator)(Eq(u, u + 1), language='C')
         assert '#pragma omp for' not in str(op4)
 
+    def test_nested_indexing(self):
+        """
+        Tests that nested functions used as indices are indexified.
+        """
+        x = Dimension(name='x')
```

Comment:
if you don't wanna use Grid (don't know why), you can also:

```
from devito import dimensions
x, y1, y2 = dimensions('x y1 y2')
```

it's a shortcut

Reply:
I will add for grid, as also noted in description tasks.

Reply:
Added some tests. Will start removing tests considered redundant soon.